File: src/CSharp/884.UncommonFromSentences.cs

using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.UncommonFromSentences
{
public class Solution {
    public string[] UncommonFromSentences(string A, string B) {
        Dictionary<string, int> hash = new Dictionary<string, int>();
        List<string> r = new List<string>();
        foreach (string s in A.Split(' ')) 
        {
            if (hash.ContainsKey(s) == false)
            {
                hash.Add(s, 1);
            }
            else
            {
                hash[s] += 1;
            }
        }
        foreach (string s in B.Split(' ')) 
        {
            if (hash.ContainsKey(s) == false)
            {
                hash.Add(s, 1);
            }
            else
            {
                hash[s] += 1;
            }
        }
        foreach (string key in hash.Keys) 
        {
            if (hash[key] == 1) {
                r.Add(key);
            }
        }
        return r.ToArray();
    }
}
}

File: src/Java/884.UncommonFromSentences.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Set;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public String[] uncommonFromSentences(String A, String B) {
        HashMap<String, Integer> hash = new HashMap<>();
        LinkedList<String> r = new LinkedList<>();
        for (String s : A.split(" ")) {
            hash.put(s, hash.getOrDefault(s, 0) + 1);
        }
        for (String s : B.split(" ")) {
            hash.put(s, hash.getOrDefault(s, 0) + 1);
        }
        for (String key : hash.keySet()) {
            if (hash.get(key) == 1) {
                r.add(key);
            }
        }
        return r.toArray(new String[r.size()]);
    }
}
